### PR TITLE
Build PSP version with cmake

### PR DESCRIPTION
**Pull request by sharkwouter**
This PR will add support for building the PSP version of SDL2 with CMake.

## Description
At the moment SDL2 can only be build for PSP with the Makefile, even though the PSP homebrew SDK does have good CMake support. ~~This PR is a draft, because the resulting library builds, but doesn't render anything when using my example application.
I'll need to debug that further before this is truly ready.~~

One thing which is nice with this is that SDL2main is currently not build-able with the Makefile for the PSP, but with cmake SDL2main will be build. There are still some issues with SDL2main, but I will fix this in a next PR.

### CODE REVIEW

**Comment by sharkwouter** [comment]
I've also added a test which builds the PSP version of SDL with new commits.

File: src/render/psp/SDL_render_psp.c [hunk 1]
```diff
@@ -209,31 +208,32 @@ StartDrawing(SDL_Renderer * renderer)
...
 
-    unsigned char *data = NULL;
-    data = SDL_malloc(psp_texture->size);
+    src = (unsigned int*) psp_texture->data;
 
-    int j;
+    data = SDL_malloc(psp_texture->size);
 
-    for(j = 0; j < height; j++, blockaddress += 16)
+    for(int j = 0; j < height; j++, blockaddress += 16)
```
Comment (sezero) [inline src/render/psp/SDL_render_psp.c:230]: This is C99: please move to function start among the other local vars

File: src/render/psp/SDL_render_psp.c [hunk 2]
```diff
@@ -209,31 +208,32 @@ StartDrawing(SDL_Renderer * renderer)
...
     {
         unsigned int *block;
 
         block = (unsigned int*)&data[blockaddress];
 
-        int i;
-
-        for(i = 0; i < rowblocks; i++)
+        for(int i = 0; i < rowblocks; i++)
```
Comment (sezero) [inline src/render/psp/SDL_render_psp.c:236]: Likewise

File: src/render/psp/SDL_render_psp.c [hunk 3]
```diff
@@ -279,21 +282,19 @@ int TextureUnswizzle(PSP_TextureData *psp_texture)
 
     sceKernelDcacheWritebackAll();
 
-    int j;
+    ydst = (unsigned char *)data;
 
-    unsigned char *ydst = (unsigned char *)data;
-
-    for(blocky = 0; blocky < heightblocks; ++blocky)
+    for(int blocky = 0; blocky < heightblocks; ++blocky)
```
Comment (sezero) [inline src/render/psp/SDL_render_psp.c:287]: Likewise

File: src/render/psp/SDL_render_psp.c [hunk 4]
```diff
@@ -279,21 +282,19 @@ int TextureUnswizzle(PSP_TextureData *psp_texture)
...
     {
         unsigned char *xdst = ydst;
 
-        for(blockx = 0; blockx < widthblocks; ++blockx)
+        for(int blockx = 0; blockx < widthblocks; ++blockx)
```
Comment (sezero) [inline src/render/psp/SDL_render_psp.c:291]: Likewise

File: src/render/psp/SDL_render_psp.c [hunk 5]
```diff
@@ -279,21 +282,19 @@ int TextureUnswizzle(PSP_TextureData *psp_texture)
...
         {
             unsigned int *block;
 
             block = (unsigned int*)xdst;
 
-            for(j = 0; j < 8; ++j)
+            for(int j = 0; j < 8; ++j)
```
Comment (sezero) [inline src/render/psp/SDL_render_psp.c:297]: Likewise

**Comment by sezero** [comment]
Also see the compiler warnings in CI build log

**Comment by sharkwouter** [comment]
Thanks for the review. I've addressed the comments. I don't think it makes sense to make add many  code changes in this PR, though. The goal here is to make the existing code build with CMake.

**Comment by sharkwouter** [comment]
When merging this, feel free to squash it.

**Review by sezero — APPROVED** [review]
Looks OK to me.

Is the following OK to fix `-Wshadow` warnings?
```diff
diff --git a/src/joystick/psp/SDL_sysjoystick.c b/src/joystick/psp/SDL_sysjoystick.c
index a3b9bc5..2dfa8aa 100644
--- a/src/joystick/psp/SDL_sysjoystick.c
+++ b/src/joystick/psp/SDL_sysjoystick.c
@@ -63,10 +63,10 @@ static point c = { 78, 32767 };
 static point d = { 128, 32767 };
 
 /* simple linear interpolation between two points */
-static SDL_INLINE void lerp (point *dest, point *a, point *b, float t)
+static SDL_INLINE void lerp (point *dest, point *pt_a, point *pt_b, float t)
 {
-    dest->x = a->x + (b->x - a->x)*t;
-    dest->y = a->y + (b->y - a->y)*t;
+    dest->x = pt_a->x + (pt_b->x - pt_a->x)*t;
+    dest->y = pt_a->y + (pt_b->y - pt_a->y)*t;
 }
 
 /* evaluate a point on a bezier-curve. t goes from 0 to 1.0 */
```

**Comment by sharkwouter** [comment]
Thanks! That would be okay with me. You should be able to push to my branch. Let me know if you'd prefer me to add it, that's not a problem either.

**Comment by 1bsyl** [comment]
By the way, 
the PSP build was broken before 2.0.18 
https://github.com/libsdl-org/SDL/commit/dd9b2daf8dd5bfa4ccdddf1f740aea712cb2e369#diff-62c16cccfbaa41dbb3f8c91dda3e5981846fd3293100786a6faa3d73a590646f
so I am not sure how much this is used,
and maybe it needs more attention ...

Then, when added RenderGeometry:
https://github.com/libsdl-org/SDL/commit/f080273acd599a4554ecaa04dd5e59848c711905#diff-62c16cccfbaa41dbb3f8c91dda3e5981846fd3293100786a6faa3d73a590646f

maybe we could simplify and removed RenderCopy RenderCopyEx, RenderFillRect as in other backends.
(RenderGeometry, can be used to Fill, Copy and CopyEx)
But I am not sure about, because there is swizzling texture,  and if this is as fast as rect copies.

**Comment by sezero** [comment]
Merged the PR, applied the `-Wshadow` fix separately.

@sharkwouter: What do you say about @1bsyl comments?
@1bsyl: Should we create a new ticket for those?

**Comment by 1bsyl** [comment]
@sezero yes, I can create the ticket !

**Comment by sezero** [comment]
@sharkwouter: you may want to look at https://github.com/libsdl-org/SDL/issues/5075

**Comment by sharkwouter** [comment]
@1bsyl SDL2 for the PSP is being used and has seen some fixes down stream. The community has recently done some major refactors to the PSP toolchain in general and we'd like to contribute to upstream. Personally I'm more of a tools guy and user of SDL than an SDL developer, though. I don't really understand the rendering code. Thanks for making the issue. I'll see what else I can fix and ask others to help out.

**Comment by 1bsyl** [comment]
@sharkwouter thanks for the info.
The rendering code there isn't a big deal, just decide if RenderCopy, CopyEx, Fill rect needs to remains or could be removed.
maybe just forward, also for the downstream fix can be merged !

**Pull request merged**